spec/integration/autobahn_intg_spec.rb

```diff
@@ -319,6 +319,20 @@ def create_transport_system(name, options={})
         await_delivery
         @queues.map { |q| q.status.first }.reduce(:+).should == 1
       end
+
+      it 'does not deliver any new messages after it has been unsubscribed' do
+        unsubscribed = Autobahn::Concurrency::Semaphore.new(0)
+        @consumer.subscribe do |headers, message|
+          begin
+            unsubscribed.acquire
+            headers.ack
+          rescue java.lang.InterruptedException
+          end
+        end
+        unsubscribed.release
+        @consumer.unsubscribe!
+        unsubscribed.has_queued_threads.should be_false
+      end
```

Comment:
Yeah, the test itself was really horrible and as you say, tried to do everything at once. I like your suggestions and think I managed to cover them all.